File: week-6/go/membersys/repository/member_repository.go
package repository

import (
	. "Project/membersys/core"
)

type MemberRepository interface {
	AddMember(name string, username string, password string) bool
	GetMember(username string, password string) Member
}

func AddMember(r MemberRepository, name string, username string, password string) bool {
	return r.AddMember(name, username, password)
}

func GetMember(r MemberRepository, username string, password string) Member {
	return r.GetMember(username, password)
}

File: week-7/py/member_system/repository/member_repository.py
from member_system.core import Member

class MemberRepository:
    def addMember(self, __name: str, __username: str, __password: str) -> bool:
        return NotImplemented

    def getMember(self, __username: str, __password: str) -> Member | None:
        return NotImplemented

    def getMemberByUsername(self, __username: str) -> Member | None:
        return NotImplemented

    def updateNameById(self, __id: int, __newName: str) -> bool:
        return NotImplemented
